src/components/List.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+    <div>
+        <div v-if="isEditMode" class="addArea">
+            <p>Edit mode is activated!</p>
+            <input text="text" v-model="updatedCity.name" />
+            <button @click="changeMode(),updateCity(updatedCity)">Save</button>
+            <button @click="changeMode()">Cancel</button>
+        </div>
+
+        <div v-else class="addArea">
+            <p>Normal mode is on!</p>
+            <input type="text" v-model="newCity.name" />
+            <button @click="addCity(Object.assign({}, newCity)),resetInput()">Add</button>
```

Comment:
Same here, complex logic should go into a method. Instead of `Object.assign(something)` you could use `{...newCity}` this will create a copy of that object